pkg/api/mlflow/controller/artifact.go

```diff
@@ -26,3 +61,53 @@ func (c Controller) ListArtifacts(ctx *fiber.Ctx) error {
 	log.Debugf("artifactList response: %#v", resp)
 	return ctx.JSON(resp)
 }
+
+// GetArtifact handles `GET /artifacts/get` endpoint.
+func (c Controller) GetArtifact(ctx *fiber.Ctx) error {
+	req := request.GetArtifactRequest{}
+	if err := ctx.QueryParser(&req); err != nil {
+		return api.NewBadRequestError(err.Error())
+	}
+	log.Debugf("GetArtifact request: %#v", req)
+
+	artifact, err := c.artifactService.GetArtifact(ctx.Context(), &req)
+	if err != nil {
+		return err
+	}
+
+	filename := getFilename(req.Path)
+	ctx.Set("Content-Type", getContentType(filename))
+	ctx.Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s", filename))
+	ctx.Set("X-Content-Type-Options", "nosniff")
+	ctx.Context().Response.SetBodyStreamWriter(func(w *bufio.Writer) {
+		defer artifact.Close()
+		bytesWritten, err := io.CopyBuffer(w, artifact, make([]byte, 4096))
+		if err != nil {
+			log.Errorf("error encountered in %s %s: error streaming artifact: %s", ctx.Method(), ctx.Path(), err)
+		}
+		log.Debugf("GetArtifact wrote bytes to output stream: %d", bytesWritten)
+		w.Flush()
+	})
+
+	return nil
+}
+
+// getFilename returns the final bit of the path (the filename).
+func getFilename(path string) string {
+	pathParts := strings.Split(path, "/")
+	return pathParts[len(pathParts)-1]
+}
+
+// getContentType will determine the content type of the file.
+func getContentType(filename string) string {
+	fileParts := strings.Split(filename, ".")
+	fileExt := fileParts[len(fileParts)-1]
+	if slices.Contains(textTypes, fileExt) {
+		return "text/plain"
+	}
+	mimeType := mime.TypeByExtension("." + fileExt)
+	if mimeType != "" {
+		return mimeType
+	}
+	return "application/octet-stream"
+}
```

Comment:
can you please use golang `path` package?  also please move this functions to `helpers.go` to keep `controller` clean.

Reply:
i'm not sure we should move this unless it is needed by a second controller -- it forces the methods to become public, moves them away from the place they are used, etc

Reply:
but please, this is not about to use it in several places. right now it uses only in one place so it means that we even don't need to move this logic into the function, right? this is a bout organisation in our project and we try to keep such unrelated logic in `helpers.go` throughout the project.

Reply:
went ahead and changed to move this along